mongo/client_test.go

```diff
@@ -147,6 +147,74 @@ func TestClient(t *testing.T) {
 		client := setupClient(options.Client().SetReadConcern(rc))
 		assert.Equal(t, rc, client.readConcern, "expected read concern %v, got %v", rc, client.readConcern)
 	})
+	t.Run("heartbeatfrequencyms from SetHeartbeatInterval()", func(t *testing.T) {
```

Comment:
Suggest removing these constructor-level tests as they duplicate the option validation test.